File: lessons/compiled_lang/cs/src/e60.branch.cs
using System;
using System.Text.RegularExpressions;

class E60Branch
{
    static void Main()
    {
        Console.Write("Input a character: ");
        char keypress = (char)Console.Read();
        string s = keypress.ToString();

        if (Regex.IsMatch(s, "[A-Z]"))
            Console.WriteLine("Uppercase letter");
        else if (Regex.IsMatch(s, "[a-z]"))
            Console.WriteLine("Lowercase letter");
        else if (Regex.IsMatch(s, "[0-9]"))
            Console.WriteLine("Digit");
        else
            Console.WriteLine("Punctuation, whitespace, or other");
    }
}

File: lessons/compiled_lang/cpp/src/e60.branch.cpp
#include <iostream>
#include <regex>
#include <string>

int main() {
    std::cout << "Input a character: ";
    char keypress;
    std::cin.get(keypress);
    std::string s(1, keypress);

    if (std::regex_search(s, std::regex("[A-Z]")))
        std::cout << "Uppercase letter" << std::endl;
    else if (std::regex_search(s, std::regex("[a-z]")))
        std::cout << "Lowercase letter" << std::endl;
    else if (std::regex_search(s, std::regex("[0-9]")))
        std::cout << "Digit" << std::endl;
    else
        std::cout << "Punctuation, whitespace, or other" << std::endl;

    return 0;
}
